deploy: Improve deployment

juliaogris opened the pull request:
Update kubecfg version to camh's fork[1], as the bitnami/kubecfg version
doesn't support remote sources which we need for jcdc, e.g.:

   kubecfg show https://github.com/foxygoat/foxtrot/raw/master/deployment/main.jsonnet

Add k8s service account and cluster role (binding).

Sneak in minor Makefile changes:

- better help regexp
- updated DOCKER_TAG variable
- undeploy diff-deploy targets

[1]: https://github.com/camh-/kubecfg/commit/7f47c390f509d8be9e6e3adf60c9e372c6a2589b

camh- reviewed (dismissed):
LGTM :herb:

camh- commented on Makefile:
```diff
@@ -103,7 +103,7 @@ docker-test: docker-build
 # --- Deployment -------------------------------------------------------------------
 TLA_ARGS = \
 	$(if $(DEPLOY_HOSTNAME), --tla-str hostname=$(DEPLOY_HOSTNAME)) \
-	--tla-str docker_tag=$(DOCKER_TAG)) \
```
looks like this should be fixed in the previous commit

camh- replied:
previous previous commit (the first one)

juliaogris committed:
make: Update make help

Update make help to also display pattern rules. The help regexp for
matching targets now also includes `%` as we want to display help
messages for  `make deploy-%` etc.

juliaogris committed:
make: Use DOCKER_TAG instead of DEPLOY_TAG

Use DOCKER_TAG instead of DEPLOY_TAG as we already have logic on CI to
create DOCKER_TAG and also because a missing DOCKER_TAG errors - really
we never want to deploy `latest` as it doesn't get updated even if
latest has been updated.

juliaogris committed:
make: Add diff-deploy-% and undeploy-% targets

Add diff-deploy-% and undeploy-% targets to Makefile. E.g.
`diff-deploy-goat` shows the difference of between deployed and "file"
k8s manifests. `undeploy-goat` deletes the deployment.

juliaogris committed:
docker: Update kubecfg version

Update kubecfg version to camh's fork[1], as the bitnami/kubecfg version
doesn't support remote sources which we need for jcdc, e.g.:

        kubecfg show https://github.com/foxygoat/foxtrot/raw/master/deployment/main.jsonnet

[1]: https://github.com/camh-/kubecfg/commit/7f47c390f509d8be9e6e3adf60c9e372c6a2589b

juliaogris committed:
deploy: Add service account with cluster role

Add three new k8s resources: jcdc service account, so that jcdc can
redeploy other pods, a cluster role, which gives the priveleges to do so
and a cluster role binding which gives these privileges to jcdc service
account.

juliaogris committed:
deploy: Update image pull policy

Update image pull policy such that image tags starting with `pr` will
always be re-pulled. This is a preparatory step for per-branch
deployments, so that when a pull request gets updated a new image gets
pulled even if the label (prNN) hasn't changed.

camh- reviewed (approved):
LGTM :herb:
